test/parallel_api/numeric/numeric.ops/exclusive_scan.pass.cpp

```diff
@@ -0,0 +1,104 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+#include _PSTL_TEST_HEADER(numeric)
+
+#include "support/utils.h"
+
+#include <numeric>
+
+#if TEST_DPCPP_BACKEND_PRESENT
+
+#include "support/sycl_alloc_utils.h"
+
+template <sycl::usm::alloc alloc_type>
+void
+test_with_usm(sycl::queue& q, const ::std::size_t count)
+{
+    auto prepare_data = [count](std::vector<::std::size_t>& idx, std::vector<int>& val)
+    {
+        for (int i = 0; i < count; i++)
+        {
+            idx[i] = i + 1;
+            val[i] = 0;
+        }
+    };
+
+    // Prepare source data
+    std::vector<::std::size_t> h_idx(count);
+    std::vector<int>           h_val(count);
+    prepare_data(h_idx, h_val);
+
+    // Copy source data to USM shared/device memory
+    TestUtils::usm_data_transfer<alloc_type, ::std::size_t> dt_helper_h_idx(q, ::std::begin(h_idx), ::std::end(h_idx));
+    auto d_idx = dt_helper_h_idx.get_data();
+
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper_h_val(q, ::std::begin(h_val), ::std::end(h_val));
+    auto d_val = dt_helper_h_val.get_data();
+
+    // Run dpl::exclusive_scan algorithm on USM shared-device memory
+    auto myPolicy = oneapi::dpl::execution::make_device_policy<
+        TestUtils::unique_kernel_name<class copy, (::std::size_t)alloc_type>>(q);
+    oneapi::dpl::exclusive_scan(myPolicy, d_idx, d_idx + count, d_val, 0);
+
+    // Copy results from USM shared/device memory to host
+    std::vector<::std::size_t> h_sidx(count);
+    std::vector<int>           h_sval(count);
+    dt_helper_h_idx.retrieve_data(h_sidx.begin());
```

Comment:
Why do we need the source data copying back?
Actually we should check just result.

Reply:
Fixed